### PR TITLE
[MOBILE-522-iOS] Unity: Clean up cruft in “clean” step

### PR DESCRIPTION
**Pull request by BrianBatchelder**
### What do these changes do?
Fix migration to new version of the plugin for iOS projects. The upgrade process now completes without errors:

1. Download new version of the plugin (urbanairship-X.Y.Z.unitypackage) from bintray.
2. Open existing project in Unity
3. In Unity:
   1. Upgrade to new plugin (As of Unity 2019.1.1f1: "Assets" → "Import Package" → "Custom Package")
   1. Build and Run (As of Unity 2019.1.1f1: "File" → "Build Settings")
      1. If necessary, switch to iOS Platform
      1. Build and Run
      1. _Unity launches Xcode_
8. In Xcode:
   1. if necessary, select a development team.
   9. Run

### Why are these changes necessary?
Developers have reported issues upgrading to new versions of our plugin. 

### How did you verify these changes?
Lots of upgrades of the ua-sample-game project

### Anything else a reviewer should know?
* There was existing migration code. It was just not up-to-date.
* New migration code used existing patterns, or patterns from [GPGSUpgrader.cs](https://github.com/playgameservices/play-games-plugin-for-unity/blob/d8d3267db13878c5e3fefb69c516e15151eba531/source/PluginDev/Assets/GooglePlayGames/Editor/GPGSUpgrader.cs), which is what UAUpdater was originally based on.
* I focused on cross-platform configuration and iOS platform-specific migration for this first PR. Android platform-specific migration will follow.

### CODE REVIEW

**Comment by crow** [comment]
🍔 lgtm

File: Assets/UrbanAirship/Editor/UAConfig.cs [hunk 1]
```diff
@@ -22,6 +22,7 @@ public enum LogLevel {
             Debug = 1,
             Info = 2,
             Warn = 3,
+            Warning = -1,
```
Comment (rlepinski) [inline Assets/UrbanAirship/Editor/UAConfig.cs]: Why do we have `Warn` and `Warning` ?

File: Assets/UrbanAirship/Editor/UAConfig.cs [hunk 2]
```diff
@@ -113,8 +114,6 @@ public enum LogLevel {
             this.NotificationPresentationOptionBadge = config.NotificationPresentationOptionBadge;
             this.NotificationPresentationOptionSound = config.NotificationPresentationOptionSound;
 
-            this.ProductionFCMSenderId = config.ProductionFCMSenderId;
-            this.DevelopmentFCMSenderId = config.DevelopmentFCMSenderId;
```
Comment (rlepinski) [inline Assets/UrbanAirship/Editor/UAConfig.cs]: Why did you delete this?

File: Assets/UrbanAirship/Editor/UAConfig.cs [hunk 3]
```diff
@@ -125,25 +124,31 @@ public enum LogLevel {
...
             }
 
+            bool migratedConfig = false;
             try {
                 if (File.Exists (filePath)) {
                     using (Stream fileStream = File.OpenRead (filePath)) {
                         XmlSerializer serializer = new XmlSerializer (typeof (UAConfig));
                         UAConfig config = (UAConfig) serializer.Deserialize (fileStream);
-                        config.Migrate ();
+                        migratedConfig = config.Migrate ();
```
Comment (rlepinski) [inline Assets/UrbanAirship/Editor/UAConfig.cs]: Do you need to migrate config?

File: Assets/UrbanAirship/Editor/UAConfigEditor.cs [hunk 4]
```diff
@@ -24,14 +24,12 @@ public class UAConfigEditor : EditorWindow {
                 config.ProductionAppKey = EditorGUILayout.TextField ("App Key", config.ProductionAppKey);
                 config.ProductionAppSecret = EditorGUILayout.TextField ("App Secret", config.ProductionAppSecret);
                 config.ProductionLogLevel = (UAConfig.LogLevel) EditorGUILayout.EnumPopup ("Log level:", config.ProductionLogLevel);
-                config.ProductionFCMSenderId = EditorGUILayout.TextField ("Android FCM Sender ID:", config.ProductionFCMSenderId);
```
Comment (rlepinski) [inline Assets/UrbanAirship/Editor/UAConfigEditor.cs]: Why are you removing this

File: Assets/UrbanAirship/Editor/UAConfig.cs (hunk 1 above)
Comment (BrianBatchelder) [reply Assets/UrbanAirship/Editor/UAConfig.cs]: It was changed from "Warning" to "Warn" to match what Android calls it. But there was no migration for the config file. That was causing the old config files to fail to load, which caused us to clear the entire config. Now "Warning" loads correctly and is migrated to "Warn".

File: Assets/UrbanAirship/Editor/UAConfig.cs (hunk 3 above)
Comment (BrianBatchelder) [reply Assets/UrbanAirship/Editor/UAConfig.cs]: Yes. As indicated above.

File: Assets/UrbanAirship/Editor/UAConfig.cs (hunk 2 above)
Comment (BrianBatchelder) [reply Assets/UrbanAirship/Editor/UAConfig.cs]: Nice catch. I think it was accidentally deleted when I was working on the Android side trying to deal with google-services.json.

File: Assets/UrbanAirship/Editor/UAConfigEditor.cs (hunk 4 above)
Comment (BrianBatchelder) [reply Assets/UrbanAirship/Editor/UAConfigEditor.cs]: As above.

**Comment by rlepinski** [comment]
👍
